test/unit/sites.js

```diff
@@ -3,179 +3,325 @@
-          site_name: 'old name',
-          site_address: 'old address'
-        }, data[0]);
+        assert.deepEqual([site9], data);
 
         return sites.updateSite({
           params: {
-            site_id: 54321
+            site_id: 9
           },
-          data: {
-            updateValues: {
-              site_name: 'new name',
-              site_address: 'new address'
-            }
+          body: {
+            site_name: 'new name',
+            site_address: 'new address'
           }
         });
       })
       .then(function() {
-        return site.getSite({
+        return sites.getSite({
           params: {
-            site_id: 54321
+            site_id: 9
           }
         });
       })
       .then(function(data) {
-        assert.equal({
-          site_id: 54321,
+        assert.deepEqual([{
+          site_id: 9,
           site_name: 'new name',
           site_address: 'new address'
-        }, data[0]);
+        }], data);
         done();
       });
     });
 
-    xit('delete a site', function(done) {
+    it('does not updated a site because the body was missing', function(done) {
+      sites.updateSite({}).catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, or address');
+        done();
+      });
+    });
+
+    it('does not updated a site because there are no fields to be updated', function(done) {
+      sites.updateSite({
+        body: {}
+      })
+      .catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, or address');
+        done();
+      });
+    });
+
+    it('updates a site that does not exist', function(done) {
+      sites.updateSite({
+        params: {
+          site_id: 92342
+        },
+        body: {
+          site_name: 'new name',
+          site_address: 'new address'
+        }
+      })
+      .then(function(data) {
+        assert.equal(0, data.affectedRows);
+        done();
+      });
+    });
+  });
+
+  describe('deleteSite(req)', function() {
+    it('delete a site', function(done) {
       sites.getSite({
         params: {
-          site_id: 55555
+          site_id: 3
         }
       })
       .then(function(data) {
-        assert.equal({
-          site_id: 55555,
-          site_name: 'to delete',
-          site_address: 'Earth'
-        }, data[0]);
+        assert.deepEqual([site3], data);
 
-        return site.deleteSite({
+        return sites.deleteSite({
           params: {
-            site_id: 55555
+            site_id: 3
           }
         });
       })
       .then(function() {
-        return site.getSite({
+        return sites.getSite({
+          params: {
+            site_id: 3
+          }
+        });
+      })
+      .then(function(data) {
+        assert.deepEqual([], data);
+        done();
+      });
+    });
+
+    it('deletes a site that does not exist', function(done) {
```

Comment:
clarify: "Does nothing when told to delete a site that doesn't exist"
Or something like that